### ISSUE TITLE
Cannot Use a Reporter with no File/Stream Output (Reports to API) - Should Allow output: null

### ISSUE THREAD

**Issue by djbreen7** [DevExpress/testcafe#3765]
### What is your Test Scenario?

 Should allow `null` as output on custom reporters

### What is the Current behavior?

 Output must be a stream or file path. Fails on null.

### What is the Expected behavior?

Should allow `null` as output on custom reporters. For example, I've built a custom reporter that sends the test results to an API which takes the place of the and file or stream output

Example from pre-1.0
```js
        // This worked
        runner.reporter('toucan', null);
        runner.reporter('html', htmlReport);
        runner.reporter('teamcity');
```

### What is your web application and your TestCafe test code?

Reporter: https://github.com/djbreen7/testcafe-reporter-toucan
Runner:
```js
...

require('testcafe-reporter-toucan');
require('testcafe-reporter-html');

let testcafe = null;

createTestCafe('localhost', 1337, 1338)
    .then(tc => {
        testcafe = tc;
        const runner = testcafe.createRunner();
        const htmlReport = fs.createWriteStream('./reports/integration_report.html');
        const devNull = fs.createWriteStream('//./NUL/devnull'); // tried (see commented code block)
        const reporters = [
            { name: 'html', output: htmlReport },
            'teamcity',
            { name: 'toucan', output: null }, // Error: Specify a file name or a writable stream as the reporter's output target.
        ]

        /** Also tried:
         * { name: 'toucan', output: devnull} // Hangs
         * 'toucan' // Multiple reporters attempting to write to stdout: "toucan, teamcity". Only one reporter can write to stdout.
         */      
        
        return runner
            .screenshots('./screenshots/testsuite', true)
            .src([
                // test fixtures
            ])
            .browsers(['chrome:headless --no-sandbox'])
            .reporter(reporters)
            ...
```
 
Your website URL (or attach your complete example):

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
 
```
</details>

<details>
<summary>Your complete test report:</summary>

```

```
</details>

<details>
<summary>Screenshots:</summary>

```

```
</details>

### Steps to Reproduce:

1. Create a runner with a custom reporter that has no output
2. Add a reporter with a stream output, file output, and no output
3. Run the tests
 
### Your Environment details:
 
* testcafe version: 1.1.3                  
* node.js version: 10.15.3                   
* command-line arguments: na 
* browser name and version: chrome 73 
* platform and version: Windows 10 Pro          
* other:

**Comment by miherlosev** [comment]
Hi @djbreen7 
It's a good idea - the possibility to report the test result to output that differs to file or stream. Thank you for share this idea with us.

At present, I can suggest the following workaround for your case:
```js
const runner = testcafe.createRunner();
        const htmlReport = fs.createWriteStream('./reports/integration_report.html');
        const streamMock = {
            write: () => {},
            end: () => {}
        };
        const reporters = [
            { name: 'html', output: htmlReport },
            'teamcity',
            { name: 'toucan', output: streamMock }
        ]
```

**Comment by djbreen7** [comment]
@miherlosev Thanks. I just tried your workaround. It's having problems hanging without starting the test run, bu that's likely something I need to modify in my custom reporter. I'll dig into it more next week.

**Comment by AlexSkorkin** [comment]
@djbreen7 
 
Thank you for the update. Please keep us informed of your progress.

**Comment by djbreen7** [comment]
@AlexSkorkin @miherlosev 

Thanks again guys. The workaround is indeed working. My custom reporter was modifying old test runs rather than creating a new one. We intentionally designed it to do so in order to have a rolling report for each sprint

**Comment by AndreyBelym** [comment]
Closed by #4712.
